src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<? extends Machine> getAll(Class type) {
+        if (type.equals(Truck.class)) {
+            return new TruckProducer().get();
+        } else if (type.equals(Excavator.class)) {
+            return new ExcavatorProducer().get();
+        } else if (type.equals(Bulldozer.class)) {
+            return new BulldozerProducer().get();
+        }
+        return new ArrayList<>();
+    }
+
+    @Override
+    public void fill(List machines, Object value) {
+        for (int i = 0; i < machines.size(); i++) {
+            machines.set(i, value);
+        }
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> list) {
+        for (Machine l: list) {
```

Comment:
```suggestion
        for (Machine machine : list) {
```
bad naming, space before column